### PR TITLE
Added tests for timing.cpp/.h

### PR DESCRIPTION
**Pull request by TailonR**
The tests need to be reviewed

### CODE REVIEW

**Review by dgpalmieri — CHANGES_REQUESTED** [review]
If you delete `doctest.h` and `timing_test.cpp` from the `EPR/Library` directory, we should try to keep all the tests in the `EPR/Library/tests` directory as Brian mentioned.

Also, I think it's a little insane that this takes 90 seconds to run (especially without saying anything about it) - is there a way that we could bring that down to 15 seconds or so? I think that if you change from seconds to quarter seconds or 1/10th of a second in the get_time() test case (line 70) you could make the runtime significantly shorter.

I would also add some docstring style comments for the auxiliary functions - what do they do, why are they set up like they are?

I also don't understand what the point of the text files your generating are - do they just capture output? It might be good to notate that somewhere as well (perhaps in a docstring!).

File: EPR/Library/tests/timing_test.cpp [hunk 1]
```diff
@@ -0,0 +1,130 @@
+#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
+#include "doctest.h"
+#include "../timing.h"
```
[the remark this answers is not in the document]
Comment (cinderbdt) [reply EPR/Library/tests/timing_test.cpp:3]: Agreed. Makefile uses the -L($LIB) to solve this problem.

[the remark this answers is not in the document]
Comment (cinderbdt) [reply EPR/Library/tests/timing_test.cpp:3]: I updated the Makefile to use -iquote for this.  The cpp file should just `#include "timing.h"`

[the remark this answers is not in the document]
Comment (cinderbdt) [reply EPR/Library/tests/timing_test.cpp]: I tried editing the file, but I found that there are two files of different lengths so I'm not quite sure which to keep.

File: EPR/Library/tests/timing_test.cpp [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+#include <chrono>
+#include <thread>
+#include <fstream>
+#include <cstdio>
+
+/*******************************************************
+ *                 Auxillary functions                 *
```
[the remark this answers is not in the document]
Comment (cinderbdt) [reply EPR/Library/tests/timing_test.cpp]: I fixed the spelling of this word in both tests/test_timing.cpp and test_timing.cpp, but I'm not sure which one to keep.

**Review by mattp0 — APPROVED** [review]

**Review by dgpalmieri — APPROVED** [review]

**Pull request merged**